caffe2/contrib/tensorrt/tensorrt_op_trt.cc

```diff
@@ -54,14 +57,51 @@ TensorRTOp::TensorRTOp(const OperatorDef& operator_def, Workspace* ws)
   {
     auto engine_string =
         OperatorBase::GetSingleArgument<std::string>("serialized_engine", "");
-    CAFFE_ENFORCE(!engine_string.empty(), "Empty serialized TensorRT engine!");
-    auto trt_runtime = tensorrt::TrtObject(nvinfer1::createInferRuntime(logger_));
-    // TODO(support trt plugin factory)
-    trt_engine_ = tensorrt::TrtObject(trt_runtime->deserializeCudaEngine(
-        engine_string.data(), engine_string.size(), nullptr));
+    if (!engine_string.empty()) {
+      auto trt_runtime =
+          tensorrt::TrtObject(nvinfer1::createInferRuntime(logger_));
+      // TODO(support trt plugin factory)
+      trt_engine_ = tensorrt::TrtObject(trt_runtime->deserializeCudaEngine(
+          engine_string.data(), engine_string.size(), nullptr));
+    } else {
+      auto onnx_model_str =
+          OperatorBase::GetSingleArgument<std::string>("onnx_model", "");
```

Comment:
Is "onnx_model" and "serialized_engine" mutually exclusive? If so, please add an `ENFORCE`

Reply:
Actually I already added the next line...